persistent/Database/Persist/SqlBackend/Internal.hs

```diff
@@ -132,6 +133,21 @@ data SqlBackend = SqlBackend
     -- When left as 'Nothing', we default to using 'defaultRepsertMany'.
     --
     -- @since 2.9.0
+    , connVault :: Vault
+    -- ^ Carry arbitrary payloads for the connection that
+    -- may be used to propagate information into hooks.
```

Comment:
hm, ok, so the `Vault` is a per-`SqlBackend` thing. let me go review the `unsafePerformIO` stuff again